app/assets/javascripts/modules/moj.matchVisitors.js

```diff
@@ -0,0 +1,208 @@
+(function() {
+  'use strict';
+
+  moj.Modules.MatchVisitors = {
+
+    init: function() {
+      var self = this;
+
+      this.$el = $('.js-visitorList.nomis-enabled');
+      this.$noAdultMessage = this.$el.find('.js-noAdults');
+      this.$notAllMessage = this.$el.find('.js-notAllProcessed');
+      this.totalVisitors = this.$el.find('select').length;
+
+      this.$el.on('change', 'select', function() {
```

Comment:
It would be a lot more readable and maintainable to use a named function and use ```jQuery.proxy``` to avoid all the ```var self = this```.